src/main/java/org/apache/commons/imaging/formats/tiff/TiffRasterData.java

```diff
@@ -117,42 +128,37 @@ public TiffRasterStatistics getSimpleStatistics() {
      * @param valueToExclude exclude samples with this specified value.
      * @return a valid instance.
      */
-    public TiffRasterStatistics getSimpleStatistics(final float valueToExclude) {
-        return new TiffRasterStatistics(this, valueToExclude);
-    }
+    public abstract TiffRasterStatistics getSimpleStatistics(float valueToExclude);
 
     /**
-     * Gets the width (number of columns) of the raster.
+     * Returns the content stored as an array in this instance. Note that in
+     * many cases, the returned array is <strong>not</strong> a safe copy of the
+     * data but a direct reference to the member element. In such cases,
+     * modifying it would directly affect the content of the instance. While
+     * this design approach carries some risk in terms of data security, it was
+     * chosen for reasons of performance and memory conservation. TIFF images
+     * that contain floating-point data are often quite large. Sizes of 100
+     * million raster cells are common. Making a redundant copy of such a large
+     * in-memory object might exceed the resources available to a Java
+     * application.
```

Comment:
+1 for design decision aiming at performance.

@gwlucastrig I think this information is really important for users of this class. But not sure if every user will read this method's javadoc. What do you think if we added a shorter version to the top javadoc of this class?

Saying that users must be aware that methods in this class return the content stored as an array and that while that could be a security issue we chose it for performance consideration. See methods {@link getWidth} etc.... ? Or something along these lines?

Reply:
done